### PR TITLE
Add docker-compose.yaml and GitHub Action for CI

### LINKED ISSUE #27: Should we add a CI?
Do you think it will be worth adding a CI here? For testing on different archs... @bwesterb

### PR DESCRIPTION
**Pull request by Lekensteyn**
Tests can be run from the repo with: docker-compose run test

Fixes #27

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  pull_request:
+  push:
+    branches:
+      - cf
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Run tests
```
Comment (cjpatton) [inline .github/workflows/test.yml:12]:
```suggestion
      - name: Build Go and run tests
```

**Review by cjpatton — APPROVED** [review]

File: .github/workflows/test.yml (hunk 1 above)
Comment (Lekensteyn) [reply .github/workflows/test.yml:12]: I'll consider building as part of the test :)

Comment (cjpatton) [reply .github/workflows/test.yml:12]: lol, ok

**Review by claucece — APPROVED** [review]
LGTM ;)

**Pull request merged**